draft-ietf-rtcweb-mdns-ice-candidates.md

```diff
@@ -400,12 +400,20 @@ supported, or TURN relay if not. This may result in reduced connectivity,
 reduced throughput and increased latency, as well as increased cost in case of
 TURN relay.
 
+During experimental testing of the mDNS technique across a set of known
+mDNS-aware endpoints that had configured a STUN server but not a TURN server,
+the observed impact to ICE connection rate was 2% (relative) when mDNS
+was enabled on both sides, compared to when mDNS was only enabled on one
+side. In this testing, the percentage of connections that required STUN
+(i.e., went through a NAT) increased from 94% to 97%, indicating that
+mDNS succeeded about half the time, and fell back to NAT hairpin for the
```

Comment:
"When mDNS was only enabled on one side" probably means that:
- One side can generate and resolve mDNS candidates
- The other side cannot generate and cannot resolve mDNS candidates. It provides host candidates.

At least in simple networks, if one endpoint provides host candidate + STUN candidate, the other endpoint might not need to provide any of its own candidate to successfully connect. 
Or maybe I am missing something?

Also, is 2% relative to all connections, including connections that might have been granted getUserMedia (in which case the mDNS enabled endpoint might not generate any mDNS candidate)?

Reply:
> "When mDNS was only enabled on one side" probably means that:
> 
> * One side can generate and resolve mDNS candidates
> * The other side cannot generate and cannot resolve mDNS candidates. It provides host candidates.

Correct.

> 
> At least in simple networks, if one endpoint provides host candidate + STUN candidate, the other endpoint might not need to provide any of its own candidate to successfully connect.

That's what we expected as well, but we have observed that that is not the case. We suspect that some endpoints are malfunctioning upon receipt of the mDNS candidates or fail if there are no host candidates. (We observed that some versions of Firefox fail if the c= line contains a mDNS candidate.)

> Or maybe I am missing something?
> 
> Also, is 2% relative to all connections, including connections that might have been granted getUserMedia (in which case the mDNS enabled endpoint might not generate any mDNS candidate)?

No, it's specific to cases where mDNS is used on both sides (i.e. data channel only for us).

Reply:
> We suspect that some endpoints are malfunctioning upon receipt of the mDNS candidates or fail if there are no host candidates. (We observed that some versions of Firefox fail if the c= line contains a mDNS candidate.)

Maybe we should add these two statements.

Reply:
> However, some legacy endpoints are not fully spec-compliant and can behave unpredictably in the presence of ICE candidates that contain a hostname

This sentence covered part of it, but I am also thinking if we should include a statement on the c= line issue during the experimental testing; technically this is specific to the experiment before we update the draft in the other PR.

Reply:
I re-read the text here. Given that the experiment numbers remained similar even after releasing a fix for for the c= line issue, and this paragraph specifically talks about when both endpoints are mdns-aware, I am inclined to leave the text as-is.

Note also that I talk about problems in the backcompat case in the second following paragraph.

Thoughts?

Reply:
Yeah, the stats stayed similar and the rationale makes sense to me. Also, "mDNS-aware" clarifies the context for me, so lgtm.